test/multipart-disk.test.js

```diff
@@ -78,6 +80,135 @@ test('should store file on disk, remove on response', function (t) {
   })
 })
 
+test('should throw on file limit error', function (t) {
+  t.plan(5)
+
+  const fastify = Fastify()
+  t.tearDown(fastify.close.bind(fastify))
+
+  fastify.register(multipart)
+
+  fastify.post('/', async function (req, reply) {
+    t.ok(req.isMultipart())
+
+    try {
+      await req.saveRequestFiles({ limits: { fileSize: 500 } })
+      reply.code(200).send()
+    } catch (error) {
+      t.true(error instanceof fastify.multipartErrors.RequestFileTooLargeError)
+      t.equal(error.part.fieldname, 'upload')
+      reply.code(500).send()
+    }
+  })
+
+  fastify.listen(0, async function () {
```

Comment:
passing an async function here would swallow the error. Use async/await (see my other PR for a fix).

Reply:
Why is it ok to swallow the error?

Reply:
it is not ok. The error is being swallowed because you pass an async function in a place that's not meant to receive an async function.